### PR TITLE
cargo-readme: init at 3.2.0

### PR DESCRIPTION
**Pull request by baloo**
###### Motivation for this change

Adds cargo-readme to nixpkgs.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/rust/cargo-readme/default.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
+{ lib, stdenv, rustPlatform, fetchFromGitHub }:
+
+rustPlatform.buildRustPackage rec {
+  pname = "cargo-readme";
+  version = "3.2.0";
+
+  src = fetchFromGitHub {
+    owner = "livioribeiro";
+    repo = pname;
+    rev = "cf66017c0120ae198210ebaf58a0be6a78372974";
```
Comment (fabaff) [inline pkgs/development/tools/rust/cargo-readme/default.nix:10]: Upstream tagged the releases till 3.1.2. Looks like that it was not done for 3.2.0.

Could you please poke upstream to do it?

File: pkgs/development/tools/rust/cargo-readme/default.nix [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    sha256 = "sha256-/ufHHM13L83M3UYi6mjdhIjgXx7bZgzvR/X02Zsx7Fw=";
+  };
+
+  cargoSha256 = "sha256-QVRl6xCvztWi5zAs3PXYR4saTqO5nTBPIjdlMiMXFTM=";
+
+  patches = [
+    ./unused_return.patch
```
Comment (fabaff) [inline pkgs/development/tools/rust/cargo-readme/default.nix:17]: Could you add a link to the upstream issue if possible please? Will make it easier to track.

File: pkgs/development/tools/rust/cargo-readme/default.nix [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  ];
+
+  meta = with lib; {
+    description = "Generate README.md from docstrings";
+    homepage = "https://github.com/livioribeiro/cargo-readme";
+    license = licenses.mit;
```
Comment (fabaff) [inline pkgs/development/tools/rust/cargo-readme/default.nix:23]: The code is dual-licensed under [MIT](https://github.com/livioribeiro/cargo-readme/blob/master/LICENSE-MIT) and [Apache Software License 2.0](https://github.com/livioribeiro/cargo-readme/blob/master/LICENSE-APACHE).

```suggestion
    license = with licenses; [ mit asl20 ];
```

**Review by fabaff — CHANGES_REQUESTED** [review]

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 109591` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>cargo-readme</li>
  </ul>
</details>

File: maintainers/maintainer-list.nix [hunk 4]
```diff
@@ -970,6 +970,12 @@
     email = "sivaraman.balaji@gmail.com";
     name = "Balaji Sivaraman";
   };
+  baloo = {
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:973]: Please put this into a separate commit with the message ``maintainers: add baloo``.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 109591` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>cargo-readme</li>
</ul>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 109591` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>cargo-readme</li>
</ul>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 109591` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>1 package failed to build and are new build failure:</summary>
  <ul>
<li>cargo-readme: log was empty</li>
  </ul>
</details>

```
these derivations will be built:
  /nix/store/1g35xl0vr1jbisswq323f22szl9ky1z6-060f2daaa2b2cf981bf490dc36bcc6527545ea03.patch.drv
  /nix/store/0cd3551xrkk6svmyqvx2zc61py4i0b2w-cargo-readme-3.2.0.drv
building '/nix/store/1g35xl0vr1jbisswq323f22szl9ky1z6-060f2daaa2b2cf981bf490dc36bcc6527545ea03.patch.drv'...

trying https://github.com/livioribeiro/cargo-readme/commit/060f2daaa2b2cf981bf490dc36bcc6527545ea03.patch
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   886  100   886    0     0   2009      0 --:--:-- --:--:-- --:--:--  2013
hash mismatch in fixed-output derivation '/nix/store/n0x966l8nm7p5z44px1qgkl4z1fmyljl-060f2daaa2b2cf981bf490dc36bcc6527545ea03.patch':
  wanted: sha256:114laqw68b7zpsh99skr66lam64ls6r6rqidp5z443i2i2353w75
  got:    sha256:1swzgwhm4iwsxi04dbg3bkxs68zfm65c5v4rzi0anfix6a0hhl62
cannot build derivation '/nix/store/0cd3551xrkk6svmyqvx2zc61py4i0b2w-cargo-readme-3.2.0.drv': 1 dependencies couldn't be built
error: build of '/nix/store/0cd3551xrkk6svmyqvx2zc61py4i0b2w-cargo-readme-3.2.0.drv' faile
```
